src/tools/cephfs/JournalTool.cc

```diff
@@ -866,8 +866,11 @@ int JournalTool::recover_dentries(
         // Compose: Dentry format is dnfirst, [I|L], InodeStore(bare=true)
         bufferlist dentry_bl;
         encode(fb.dnfirst, dentry_bl);
-        encode('I', dentry_bl);
+        encode('i', dentry_bl);
```

Comment:
why is the case changed here?

Reply:
When alternate_name was introduced: 39f3440e36f1d1779861e32300128915b8047b66

We couldn't change the encoding/decoding since there was no versioning. The idea was to use a new marker for the versioned dentries (plus inode), `i`.

Reply:
Thanks for sharing that, really helpful to read through!